helper/helper_test.go

```diff
@@ -98,4 +229,126 @@ var _ = Describe("helper tests", func() {
 			Expect(isSubset).To(Equal(false))
 		})
 	})
+
+	Describe("doRestoreAgent Mocked unit tests", func() {
+		var (
+			save_singleDataFile  bool
+			save_content         int
+			save_oidFile         string
+			save_isResizeRestore bool
+			save_origSize        int
+			save_destSize        int
+			save_pipeFile        string
+			save_onErrorContinue bool
+		)
+
+		BeforeEach(func() {
+			save_singleDataFile = *singleDataFile
+			save_content = *content
+			save_oidFile = *oidFile
+			save_isResizeRestore = *isResizeRestore
+			save_origSize = *origSize
+			save_destSize = *destSize
+			save_pipeFile = *pipeFile
+			save_onErrorContinue = *onErrorContinue
+
+			*singleDataFile = false
```

Comment:
Can we just do this setup and get rid of the `save_*` vars?

Reply:
The question is still open - can we just leave the setup below
```
			*singleDataFile = false
			*content = 1
			*oidFile = "testoid.dat"
			*isResizeRestore = true
			*origSize = 5
			*destSize = 3
			*pipeFile = "mock"
			*onErrorContinue = true
```

and remove all `save*` vars (and also remove `AfterEach"()`)?

Reply:
This is not separated executable for this test, it's shared with other tests. We can not be sure about an order in which tests are started so, as long global variables are used, is it's safer to save the state after test in the state it was before us.
Use of InitializeGlobals() also doesn't look a good idea - besides global variables it initializes some staff like CleanupGroup, logging. 
So, answering your question - I prefer to have this setup to not make further changes to system under test.

Reply:
- in these init/deinit functions we setup only 8 global configs.
 - we can't predict the order of test execution
 - these init/deinit functions will work only within the "doRestoreAgent Mocked unit tests" group.
 
 So, if some test changes some config not covered by the 8 configs above (or just tests from other groups are executed), the subsequent test can become flaky (depending on the order of the tests).

In this case I suggest to move the `BeforeEach()` to the top level node, and do full setup of configs there. So we are sure that before each test we have consistent start conditions.

Reply:
okay